java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -323,12 +323,38 @@ public void testBitVector() {
       // Ensure unallocated space returns 0
       assertEquals(0, accessor.get(3));
 
+      // unset the previously set bits
       m.set(1, 0);
       m.set(1022, 0);
-
+      // this should set all the array to 0
       assertEquals(1024, accessor.getNullCount());
 
+      // set all the array to 1
       for (int i = 0; i < 1024; ++i) {
+        assertEquals(1024 - i, accessor.getNullCount());
+        m.set(i, 1);
+      }
+
+      assertEquals(0, accessor.getNullCount());
+
+      // re-allocate the vector with a new (odd) size, smaller by more than a byte than the original vector
+      // this allows to ensure that we use one byte less memory than the previous one
+      vector.allocateNew(1015);
+
+      // ensure it has been zeroed
+      assertEquals(1016, accessor.getNullCount());
```

Comment:
this is a question for @julienledem or @jacques-n 

also I would recommend asking such questions on JIRA or the mailing list instead of Slack (which is not an official channel for development of the project)